app/models/contract.rb

```diff
@@ -86,9 +87,11 @@ def self.issued_by(issuer, *ids)
     where.has { name.op('COLLATE', sql(collate)).matches(pattern)}
   }
 
-  scope :by_account, lambda { |account| where({ :user_account_id => account.id } ) }
+  scope :by_account, ->(account) { where.has { user_account_id == account } }
   scope :by_account_query, lambda { |query| where( { :user_account_id => Account.buyers.search_ids(query) } ) }
 
+  scope :paid_since, ->(paid_time = MAX_UNPAID_TIME.ago) { where.has { (paid_until >= paid_time) | (variable_cost_paid_until >= paid_time) } }
```

Comment:
Rename it to `at_least_paid_until`?

Reply:
mmmm that name sounds to me as the opposite of what we want to do...
`paid_until` means the last time the customer paid, so `paid until this day` means that day is the last day that paid. So what we want to check here is `did he pay since/from this day of the param?`. `since` in the sense of `from this day to the future`, just like we do in the regular speech.

So it could be also `paid_since_at_least` for example, but `at_least_paid_until` sounds to me like `did he pay until this day of the param?` or in another words: `did he pay the previous days of this date of the param?`
WDYT? @hallelujah

Reply:
Speaking with Hery, for now it will be `has_paid_on?`